x-pack/test/functional/apps/dataset_quality/dataset_quality_summary.ts

```diff
@@ -0,0 +1,130 @@
+      await synthtrace.clean();
+    });
+
+    it('shows poor, degraded and good count', async () => {
+      const summary = await PageObjects.datasetQuality.parseSummaryPanel();
+      expect(summary).to.eql({
+        datasetHealthPoor: '0',
+        datasetHealthDegraded: '0',
+        datasetHealthGood: '3',
+        activeDatasets: '0 of 3',
+        estimatedData: '0 Bytes',
+      });
+    });
+
+    it('updates the poor count when degraded docs are ingested', async () => {
+      // Index malformed document with current timestamp
+      await synthtrace.index(
+        getLogsForDataset({
+          to: Date.now(),
+          count: 1,
+          dataset: datasetNames[2],
+          isMalformed: true,
+        })
+      );
+
+      await browser.refresh();
+
+      await retry.try(async () => {
+        const summary = await PageObjects.datasetQuality.parseSummaryPanel();
+        const { estimatedData, ...restOfSummary } = summary;
+        expect(restOfSummary).to.eql({
+          datasetHealthPoor: '1',
+          datasetHealthDegraded: '0',
+          datasetHealthGood: '2',
+          activeDatasets: '1 of 3',
+        });
+      });
+    });
+
+    it('updates the degraded count when degraded docs are ingested', async () => {
+      // Index malformed document with current timestamp
+      await synthtrace.index(
+        getLogsForDataset({
+          to: Date.now(),
+          count: 1,
+          dataset: datasetNames[1],
+          isMalformed: true,
+        })
+      );
+
+      // Index healthy documents
+      await synthtrace.index(
+        getLogsForDataset({
+          to: Date.now(),
+          count: 10,
+          dataset: datasetNames[1],
+          isMalformed: false,
+        })
+      );
+
+      await browser.refresh();
+
+      await retry.try(async () => {
+        const { estimatedData, ...restOfSummary } =
+          await PageObjects.datasetQuality.parseSummaryPanel();
+        expect(restOfSummary).to.eql({
+          datasetHealthPoor: '1',
+          datasetHealthDegraded: '1',
+          datasetHealthGood: '1',
+          activeDatasets: '2 of 3',
+        });
+      });
+    });
+
+    it('updates active datasets and estimated data KPIs', async () => {
+      const { estimatedData: existingEstimatedData } =
+        await PageObjects.datasetQuality.parseSummaryPanel();
+
+      // Index document at current time to mark dataset as active
+      await synthtrace.index(
+        getLogsForDataset({
+          to: Date.now(),
+          count: 4,
+          dataset: datasetNames[0],
+          isMalformed: false,
+        })
+      );
+
+      await browser.refresh();
+
+      await retry.try(async () => {
+        const { activeDatasets: updatedActiveDatasets, estimatedData: updatedEstimatedData } =
+          await PageObjects.datasetQuality.parseSummaryPanel();
+
+        expect(updatedActiveDatasets).to.eql('3 of 3');
+        expect(updatedEstimatedData).to.not.eql(existingEstimatedData);
+      });
```

Comment:
Ideally after page refresh, you need to wait for summary panel to be re-rendered. Usually it is possible to add a `data-test-subj` or just your own attribute with dynamic value based on component loading status:

<img width="1206" alt="Screenshot 2024-03-07 at 18 40 26" src="https://github.com/elastic/kibana/assets/10977896/2a87a7c9-798b-4432-b05a-9464cade97b7">

e.g. `data-test-subj="summaryPanel-${isLoaded ? 'loaded' : 'loading'})"` and later have add a PO function like
```
async waitForSummaryPanelToBeLoaded() {
    await testSubjects.find('summaryPanel-loaded');
}
```

and fetch data without retry.  This way validation will spot wrong re-rendering/glitch and test will be overall more stable. Thoughts?